webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/reports/ReportingService.scala

```diff
@@ -112,12 +121,22 @@ trait ReportingService {
       reports
     } else {
       val n1 = System.currentTimeMillis
-      val result = reports.view.mapValues { status =>
+      val result = reports.view.mapValues { case status =>
         NodeStatusReport.filterByRules(status, ruleIds)
-      }.filter { case (k,v) => v.reports.nonEmpty || v.overrides.nonEmpty }
+      }.filter { case (_,v) => v.reports.nonEmpty || v.overrides.nonEmpty }
       val n2 = System.currentTimeMillis
       TimingDebugLogger.trace(s"Filter Node Status Reports on ${ruleIds.size} in : ${n2 - n1}ms")
       result
     }.toMap
   }
+
+  def complianceByRules(report: NodeStatusReport, ruleIds: Set[RuleId]): ComplianceLevel = {
```

Comment:
this should return an option: if NodeStatusReport.reports is empty &nd v.overrides is empty

Reply:
actually, we are only speaking of compliance, so it would always return an empty compliance in this case, which is what we want (if saves a lot of troubles on related methods)